python/paddle/fluid/tests/unittests/gradient_checker.py

```diff
@@ -736,7 +748,9 @@ def fail_test(msg):
         v.stop_gradient = False
         v.persistable = True
     y = _as_list(y)
-
+    for u in y:
```

Comment:
fix this in right way

Reply:
done

Reply:
Due to the inconsistency of the stop_gradient behavior of dynamic graphs and static graphs,we should consider refactoring the dynamic graph unit test framework in the future.